src/Http/Controllers/CP/Assets/BrowserController.php

```diff
@@ -90,7 +90,7 @@ public function search(Request $request, $container)
             ? $container->searchIndex()->ensureExists()->search($request->search)
             : $container->queryAssets()->where('path', 'like', '%'.$request->search.'%');
 
-        $assets = $query->paginate(30);
+        $assets = $query->paginate(request('perPage', config('statamic.cp.pagination_size')));
```

Comment:
I think `request('perPage')` is probably enough without the fallback. Pretty sure it always gets sent up. (We do it that way in the entry listing)